ubg/src/search/AllFilters.js

```diff
@@ -0,0 +1,189 @@
+import React, {useEffect} from 'react';
+import {makeStyles} from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import Button from '@material-ui/core/Button';
+import Filter from './Filter';
+import {useFirestore} from 'reactfire';
+import PropTypes from 'prop-types';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  button: {
+    marginTop: theme.spacing(1.5),
+    marginRight: theme.spacing(3),
+    float: 'right',
+  },
+  pagination: {
+    '& > *': {
+      margin: theme.spacing(2),
+    },
+  },
+}));
+
+/**
+ * @param {object} setPaginationCount function to set pagination
+ * @param {object} setGames function to set games
+ * @return {ReactElement} All Filters and related buttons
+ */
+export default function AllFilters({setPaginationCount, setGames}) {
+  const classes = useStyles();
+  const ref = useFirestore().collection('games');
+  const [minAge, setMinAge] = React.useState(21);
+  const [minPlayer, setMinPlayer] = React.useState(1);
+  const [maxPlayer, setMaxPlayer] = React.useState('8+');
+  const [minTime, setMinTime] = React.useState(5);
+  const [maxTime, setMaxTime] = React.useState('240+');
+  const [sortBy, setSortBy] = React.useState('rating');
+  const [initialize, setInitialize] = React.useState(false);
+  const handleFilter = () => {
+    const newGames = [];
+    let list = [];
+    let maxP = maxPlayer;
+    if (typeof maxP === 'string') {
+      maxP = Number.MAX_SAFE_INTEGER;
+    }
+    let maxT = maxTime;
+    if (typeof maxT === 'string') {
+      maxT = Number.MAX_SAFE_INTEGER;
+    }
+    if (minPlayer > maxP) {
+      setMaxPlayer(minPlayer);
+      maxP = minPlayer;
+    }
+    if (minTime > maxT) {
+      setMaxTime(minTime);
+      maxT = minTime;
+    }
+    ref.orderBy(sortBy, 'desc')
+        .get()
+        .then(function(querySnapshot) {
+          querySnapshot.forEach(function(doc) {
+            if (doc.data()['minPlayer'] <= maxP &&
+            minPlayer <= doc.data()['maxPlayer'] &&
+            doc.data()['minPlaytime'] <= maxT &&
+            minTime <= doc.data()['maxPlaytime'] &&
+            doc.data()['minAge'] <= minAge) {
+              list.push(doc.data());
+              if (list.length === 12) {
+                newGames.push(list);
+                list = [];
+              }
+            }
+          });
+          if (list.length !== 0) {
+            newGames.push(list);
+          }
+          setGames(newGames);
+          setPaginationCount(newGames.length);
+        })
+        .catch(function(error) {
+          console.log('Error getting documents: ', error);
+        });
+  };
+  const handleClear = () => {
+    setMinAge(21);
+    setMinPlayer(1);
+    setMaxPlayer('8+');
+    setMinTime(5);
+    setMaxTime('240+');
+    const newGames = [];
+    let list = [];
+    ref.orderBy(sortBy, 'desc')
+        .get()
+        .then(function(querySnapshot) {
+          setInitialize(true);
```

Comment:
I think this is actually not needed, since initialize would already be true, and setting it again would cause the useEffect hook to run again.